### PR TITLE
Track advise status using GitHub issues

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

https://github.com/thoth-station/kebechet/issues/872

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

This pull request makes asking for a thoth-advise explicit as well as tracking the progress using a GitHub issue

### CODE REVIEW

**Comment by fridex** [comment]
> This pull request makes asking for a thoth-advise explicit as well as tracking the progress using a GitHub issue

Do we want to track the advise progress using GitHub issues long-term?

**Comment by KPostOffice** [comment]
> > This pull request makes asking for a thoth-advise explicit as well as tracking the progress using a GitHub issue
> 
> Do we want to track the advise progress using GitHub issues long-term?

This seemed like a good way to improve Kebechet's traceability. Do you have an alternative solution?

**Comment by fridex** [comment]
> > > This pull request makes asking for a thoth-advise explicit as well as tracking the progress using a GitHub issue
> > 
> > 
> > Do we want to track the advise progress using GitHub issues long-term?
> 
> This seemed like a good way to improve Kebechet's traceability. Do you have an alternative solution?

What about keeping the "New * release" issues open? The bot can add "advise-in-progress" labels or so to keep status on GitHub (which would help trace the status). Once the advice is finished, the issue can be closed by the bot.

-> User opens an issue -> Kebechet marks "advise-in-progress" -> Kebechet closes issue once the advise is done.

Having issues for these events can be disturbing for users.

**Comment by KPostOffice** [comment]
> > > > This pull request makes asking for a thoth-advise explicit as well as tracking the progress using a GitHub issue
> > > 
> > > 
> > > Do we want to track the advise progress using GitHub issues long-term?
> > 
> > 
> > This seemed like a good way to improve Kebechet's traceability. Do you have an alternative solution?
> 
> What about keeping the "New * release" issues open? The bot can add "advise-in-progress" labels or so to keep status on GitHub (which would help trace the status). Once the advice is finished, the issue can be closed by the bot.
> 
> -> User opens an issue -> Kebechet marks "advise-in-progress" -> Kebechet closes issue once the advise is done.
> 
> Having issues for these events can be disturbing for users.

I was originally going to do something like this but an issue arises when multiple overlays are being advised. One `New Kebechet Advise` issue can result in more than one advise.

**Comment by fridex** [comment]
> I was originally going to do something like this but an issue arises when multiple overlays are being advised. One `New Kebechet Advise` issue can result in more than one advise.

What about using comments on the issue?

**Comment by KPostOffice** [comment]
> > I was originally going to do something like this but an issue arises when multiple overlays are being advised. One `New Kebechet Advise` issue can result in more than one advise.
> 
> What about using comments on the issue?

It is hard to close the issue when complete as we need some way to connect all the advise workflows to a single issue and also know when the final one is complete so that the issue can be closed

**Comment by fridex** [comment]
> > > I was originally going to do something like this but an issue arises when multiple overlays are being advised. One `New Kebechet Advise` issue can result in more than one advise.
> > 
> > 
> > What about using comments on the issue?
> 
> It is hard to close the issue when complete as we need some way to connect all the advise workflows to a single issue and also know when the final one is complete so that the issue can be closed

What about keeping track in issue comments? Kebechet would post comments to the advise issue to notify users about actions but also to keep information about its progress.

**Comment by KPostOffice** [comment]
>
> What about keeping track in issue comments? Kebechet would post comments to the advise issue to notify users about actions but also to keep information about its progress.

We don't really have a way of closing the issue though. How do we know when the last advise has completed?

**Comment by fridex** [comment]
> > What about keeping track in issue comments? Kebechet would post comments to the advise issue to notify users about actions but also to keep information about its progress.
> 
> We don't really have a way of closing the issue though. How do we know when the last advise has completed?

Just to keep this complete - a solution discussed in person 👍🏻

**Comment by KPostOffice** [comment]
We decided that the best thing to do is keep track of the number of expected advise completions and count the # of comments reporting advise results/PRs opened at the end of kebechet_run_results. If the expected # is found, then the issue can be closed.

**Comment by KPostOffice** [comment]
@fridex PTAL

File: kebechet/managers/thoth_advise/thoth_advise.py [hunk 1]
```diff
@@ -55,6 +56,18 @@
 # Github and Gitlab events on which the manager acts upon.
 _EVENTS_SUPPORTED = ["push", "issues", "issue", "merge_request"]
 
+ADVISE_ISSUE_TITLE = "Kebechet Advise"
+
+STARTED_ADVISE_COMMENT = "Started advise with id {analysis_id} for {env_name} to get current status, click [here](\
+    https://{host}/api/v1/advise/python/{analysis_id})"
```
Comment (fridex) [inline kebechet/managers/thoth_advise/thoth_advise.py:62]: We have a separate status endpoint at `https://{host}/api/v1/advise/python/{analysis_id}/status`. Both should work fine in this case though, the endpoint used in sources will return results once available. The `/status` endpoint will notify that the analysis was successful (without results). I think the result endpoint should work in this case fine 👍🏻

**Review by fridex — APPROVED** [review]
This is very nice!! 👍🏻 💯  Eager to see this in action 🥇

**Pull request merged**